### PR TITLE
mdt: init at 0.8.1

### PR DESCRIPTION
**Pull request by 0x61nas**
## Description of changes

Add [mdt (md-tui)](https://github.com/henriklovhaug/md-tui) package, which is a terminal Markdown viewer.

### Try it
```sh
nix run github:anas-contribs/nixpkgs/by-name/md-tui#mtd
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/md/mdt/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "henriklovhaug";
+    repo = "md-tui";
+    rev = "v${version}";
+    hash = "sha256-R6HTdNoPMlfC644AhGFfCLPojm3O2DXpmtSy1S9MCz4=";
+  };
+
+  cargoHash = "sha256-GgwC25mrEePAT5PlNypyPo5qy9mC+lxVwOG1YozAicw=";
+
```
Comment (sikmir) [inline pkgs/by-name/md/mdt/package.nix:19]:
```suggestion
    buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.CoreServices ];
```

File: pkgs/by-name/md/mdt/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  meta = {
+    description = "Markdown renderer in the terminal";
+    homepage = "https://github.com/henriklovhaug/md-tui";
+    changelog = "https://github.com/henriklovhaug/md-tui/releases/tag/v${version}";
+    license = lib.licenses.agpl3Only;
+    mainProgram = "mdt";
+    maintainers = with lib.maintainers; [ anas ];
+    platforms = with lib.platforms; unix ++ windows;
+    broken = stdenv.isDarwin; # needing some apple_sdk packages
```
Comment (sikmir) [inline pkgs/by-name/md/mdt/package.nix:28]:
```suggestion
```

File: pkgs/by-name/md/mdt/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, stdenv
+}:
```
Comment (sikmir) [inline pkgs/by-name/md/mdt/package.nix:5]:
```suggestion
, darwin
}:
```

File: pkgs/by-name/md/mdt/package.nix (hunk 1 above)
Comment (0x61nas) [reply pkgs/by-name/md/mdt/package.nix:19]: i can't test this, are you sure that it'll work?

Comment (sikmir) [reply pkgs/by-name/md/mdt/package.nix:19]: Yep, tested.

Comment (0x61nas) [reply pkgs/by-name/md/mdt/package.nix:20]: Added, thank you for testing.

**Comment by IldenH** [comment]
0.8.2 is out
https://github.com/henriklovhaug/md-tui/releases/tag/v0.8.2.

**Comment by 0x61nas** [comment]
@IldenH am waiting for rust 1.79

**Review by IldenH — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Comment by donovanglover** [comment]
Note: This was merged after https://github.com/NixOS/nixpkgs/pull/318492 got merged, so now there are two `md-tui`s in nixpkgs

**Pull request merged**